core/src/serde_support.rs

```diff
@@ -0,0 +1,1457 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use alloc::format;
+use alloc::string::ToString;
+use alloc::vec::Vec;
+
+use base64 as _;
+use base64::engine::general_purpose::STANDARD as BASE64_STANDARD;
+use base64::Engine;
+use dusk_bytes::Serializable;
+use hex as _;
+use serde::de::{
+    EnumAccess, Error as SerdeError, MapAccess, Unexpected, VariantAccess,
+    Visitor,
+};
+use serde::ser::SerializeStruct;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use serde_json as _;
```

Comment:
this neither

Reply:
The issue is that unused crate dependencies aren't allowed, so removing this will cause a compiler error.